### DISCUSSION TITLE
How to install and use Pinocchio's C++ library in Windows？

### DISCUSSION THREAD

**Discussion by Rofunction** [stack-of-tasks/pinocchio#2398] [Q&A] [answered]
I recently started learning this pinocchio dynamics library, and I want to use this dynamics library in a windows environment to realize some of my ideas. I don't know what operations I need to do or what libraries I need to pre-install in order to ensure that I can compile this dynamics library successfully under the windows system. I have downloaded the ZIP files of Eigen3 and Boost.Sincerely hope that everyone can provide me with effective methods.

**Comment by ManifoldFR** [comment]
What is your experience with the C++ programming language?

If you have none but still want to use C++, we would recommend you walk through a tutorial on Visual Studio C++ (the recommended compiler suite on Windows) and use an IDE to build Pinocchio (and Boost if what you have are Boost's sources and no binaries, if you do have the binaries in your zip you don't have to rebuild).
I think the dependencies you list should suffice (unless you want collision support, in which case you need hpp-fcl and its dependencies).

You don't need to compile everything actually. You can get Pinocchio, its headers, binaries, and those of its dependencies if you use the conda package manager (which is mainly known for Python packages, but you can install eigen, boost, etc with it).

If you are not sure you want to use C++ for your project, you may want to download Pinocchio's Python package through either the pip or conda package managers.

**Comment by jcarpent** [comment, marked as answer]
The easiest approach to install Pinocchio on Windows is using conda.
You can follow this [set of instructions](https://conda.io/projects/conda/en/latest/user-guide/install/windows.html) to install conda on Windows. When installed and configured, you can follow [this procedure](https://github.com/conda-forge/pinocchio-feedstock/?tab=readme-ov-file#installing-pinocchio).
Enjoy !
